pkg/api/validate/types_test.go

```diff
@@ -56,6 +56,38 @@ func TestValidatePluginVersion(t *testing.T) {
 	}
 }
 
+func TestIsValidClusterName(t *testing.T) {
```

Comment:
Instead of having seaparate testcase slices, you can merge them together and do something like this:

```
func TestIsValidClusterName(t *testing.T) {
	tests := []struct {
		clusterName string
		wantResult  bool
	}{
		{clusterName: "", wantResult: true},
		{clusterName: "k", wantResult: true},
		{clusterName: "my.cluster", wantResult: true},
		{clusterName: "has spaces", wantResult: true},
		{clusterName: "random#characters?", wantResult: true},
		{clusterName: "1234567890", wantResult: true},
		{clusterName: "0cluster", wantResult: true},
		{clusterName: "cluster-", wantResult: true},
		{clusterName: "-cluster", wantResult: true},
		{clusterName: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", wantResult: true},
		{clusterName: "verylongtoolongnameitslongerthan63characterswhoopsthiswontworkatall", wantResult: true},
		{clusterName: "k0", wantResult: true},
		{clusterName: "osa-testing", wantResult: true},
		{clusterName: "ExampleCluster111", wantResult: true},
		{clusterName: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", wantResult: true},
	}
	for _, test := range tests {
		isValid := isValidClusterName(test.clusterName)
		if isValid != test.wantResult {
			t.Errorf("IsValidClusterName(%v) expected %v, got %v", test.clusterName, test.wantResult, isValid)
		}
	}
}
```

Reply:
I just copied this to follow the style of the surrounding tests, see e.g. https://github.com/openshift/openshift-azure/blob/ccf8397a916173471a802d8703b2e7baecd4b981/pkg/api/validate/types_test.go#L59-L88